modules/core/src/main/scala/org/tessellation/l0/infrastructure/snapshot/SnapshotService.scala

```diff
@@ -2,46 +2,82 @@ package org.tessellation.l0.infrastructure.snapshot
+import eu.timepit.refined.cats.nonNegLongMonoid
+import eu.timepit.refined.types.numeric.NonNegLong
+import fs2.Stream
 import io.estatico.newtype.ops._
 
 object SnapshotService {
 
-  private val heightInterval = PosLong(2L)
-
   def make[F[_]: Async](
-    snapshotStorage: SnapshotStorage[F]
+    snapshotStorage: SnapshotStorage[F],
+    blocksQueue: Queue[F, Set[Signed[DAGBlock]]]
   ): SnapshotService[F] =
     new SnapshotService[F] {
 
-      def createSnapshot(
-        blocks: NonEmptySet[Hashed[DAGBlock]],
-        snapshots: Set[StateChannelSnapshot],
-        nextFacilitators: NonEmptySet[PeerId]
-      ): F[GlobalSnapshot] =
+      def offer(signedBlock: Signed[DAGBlock]): F[Unit] =
         for {
-          previousSnapshotHeight <- snapshotStorage.getLastSnapshotHeight
-          nextSnapshotHeight = Height(previousSnapshotHeight.coerce + heightInterval)
+          _ <- blocksQueue.offer(Set(signedBlock))
+        } yield ()
+
+      private val emptySnapshot = Stream.eval(createEmptySnapshot())
+
+      private val normalSnapshot = Stream.fromQueueUnterminated(blocksQueue).evalMap(createNormalSnapshot)
+
+      val snapshotCreation: Stream[F, GlobalSnapshot] =
+        normalSnapshot.through(switchRepeat(10.seconds, emptySnapshot))
 
-          blocksForNextSnapshot = blocks.filter { block =>
-            val blockHeight = block.signed.height.coerce
-            blockHeight > previousSnapshotHeight.coerce && blockHeight <= nextSnapshotHeight.coerce
-          }.map(_.signed)
+      def createEmptySnapshot(): F[GlobalSnapshot] =
+        for {
+          prevHeight <- snapshotStorage.getLastSnapshotHeight
+          prevSubHeight <- snapshotStorage.getLastSnapshotSubHeight
+          prevOrdinal <- snapshotStorage.getLastSnapshotOrdinal
+          nextFacilitators <- selectNextFacilitators
+          snapshots <- snapshotStorage.getStateChannelSnapshots
+        } yield
+          GlobalSnapshot(
+            ordinal = GlobalSnapshotOrdinal(prevOrdinal.value |+| NonNegLong(1L)),
+            height = prevHeight,
+            subHeight = Height(prevSubHeight.coerce[Long] + 1L),
+            blocks = Set.empty[Signed[DAGBlock]],
+            snapshots = snapshots,
+            nextFacilitators = nextFacilitators
+          )
+
+      def createNormalSnapshot(
+        blocks: Set[Signed[DAGBlock]]
+      ): F[GlobalSnapshot] =
+        for {
+          prevHeight <- snapshotStorage.getLastSnapshotHeight
+          prevOrdinal <- snapshotStorage.getLastSnapshotOrdinal
+          nextFacilitators <- selectNextFacilitators
+          snapshots <- snapshotStorage.getStateChannelSnapshots
+        } yield
+          GlobalSnapshot(
+            ordinal = GlobalSnapshotOrdinal(prevOrdinal.value |+| NonNegLong(1L)),
+            height = Height(prevHeight.coerce + 1L),
+            subHeight = Height(0L),
+            blocks = blocks,
+            snapshots = snapshots,
+            nextFacilitators = nextFacilitators
+          )
```

Comment:
what about unifying that into a common method?

```scala

def createEmptySnapshot(): F[GlobalSnapshot] =
  (snapshotStorage.getLastSnapshotHeight, snapshotStorage.getLastSnapshotSubHeight).mapN { (height, subheight) =>
    createSnapshot(Set.empty[Signed[DAGBlock]], height, Height(subheight.coerce[Long] + 1L)
    
def createNormalSnapshot(blocks: Set[Signed[DAGBlock]]): F[GlobalSnapshot] =
  snapshotStorage.getLastSnapshotHeight.flatMap { height =>
    createSnapshot(blocks, Height(height.coerce + 1L), Height(0L))

private def createSnapshot(blocks: Set[Signed[DAGBlock]], height: Height, subheight: Height): F[GlobalSnapshot] =
        for {
            prevOrdinal <- snapshotStorage.getLastSnapshotOrdinal
            nextFacilitators <- selectNextFacilitators
            snapshots <- snapshotStorage.getStateChannelSnapshots
          } yield
            GlobalSnapshot(
              ordinal = GlobalSnapshotOrdinal(prevOrdinal.value |+| NonNegLong(1L)),
              height = height,
              subHeight = subheight,
              blocks = blocks,
              snapshots = snapshots,
              nextFacilitators = nextFacilitators
            )


```

Reply:
Yep, we will get rid of the naming "normal/empty"